datastores/gossip_kv/server/lattices/mod.rs

```diff
@@ -0,0 +1,238 @@
+use std::cmp::Ordering;
+use std::collections::HashSet;
+use std::hash::Hash;
+
+use hydroflow::lattices::{IsBot, IsTop, LatticeOrd, Merge};
+
+/// A bounded set union lattice with a fixed size N.
+///
+/// Once the set reaches size N, it becomes top. The items in the set are no longer tracked to
+/// reclaim associated memory.
+#[derive(Debug, Clone, Eq)]
+
+pub struct BoundedSetLattice<T, const N: usize>
+where
+    T: Eq + Hash,
+{
+    // The set of items in the lattice with invariant:
+    // is_top => items.is_empty() ... i.e. the items are dropped when the lattice reaches top.
+    items: HashSet<T>,
+    is_top: bool,
+}
```

Comment:
Perhaps an enum with a top/full variant and a not-yet-full variant would make more invalid states unrepresentable

Reply:
True. Will look at it separately.